troi/filters.py

```diff
@@ -289,3 +290,39 @@ def read(self, inputs, debug=False):
                         results.append(r)
 
         return results
+
+
+class LatestListenedAtFilterElement(troi.Element):
+    '''
+        Remove recordings if they have been played recently
+    '''
+
+    def __init__(self, min_number_of_days=14):
+        '''
+            Filter the recordings according to latest_listened_at field in the lb metadata. 
+            If that field is None, treat it as if the user hasn't listened to this track
+            recently or at all and keep the track in the list.
+        '''
+        troi.Element.__init__(self)
+        self.min_number_of_days = min_number_of_days
+
+    @staticmethod
+    def inputs():
+        return [Recording]
+
+    @staticmethod
+    def outputs():
+        return [Recording]
+
+    def read(self, inputs, debug=False):
+
+        recordings = inputs[0]
+
+        results = []
+        now = datetime.datetime.now()
+        for r in recordings:
+            td = now - r.listenbrainz["latest_listened_at"]
```

Comment:
This may need to handle the latest_listened_at null case. If we pass recommendations to this, then this case won't happen anyway but maybe useful in mixed source playlists.